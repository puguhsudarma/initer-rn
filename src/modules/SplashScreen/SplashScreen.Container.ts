import { connect } from 'react-redux';
import SplashScreen from './SplashScreen';
import { bindActionCreators, Dispatch } from 'redux';
import { IState } from '../../types/state';

const mapStateToProps = (state: IState) => ({
    isLoggedIn: state.account.isLoggedIn,
});

const mapDispatchToProps = (dispatch: Dispatch) => {
    return bindActionCreators({}, dispatch);
};

export interface ISplashScreenReduxProps
    extends ReturnType<typeof mapStateToProps>,
        ReturnType<typeof mapDispatchToProps> {}

export default connect(mapStateToProps, mapDispatchToProps)(SplashScreen);
