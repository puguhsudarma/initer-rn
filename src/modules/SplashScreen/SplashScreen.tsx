import React from 'react';
import { StyleSheet, View } from 'react-native';
import color from '../../constants/color';
import { ISplashScreenReduxProps } from './SplashScreen.Container';

interface ISplashScreenProps extends ISplashScreenReduxProps {
    componentId: string;
}

class SplashScreen extends React.PureComponent<ISplashScreenProps> {
    render() {
        return <View style={styles.container}></View>;
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: color.TURQUOISE,
    },
});

export default SplashScreen;
