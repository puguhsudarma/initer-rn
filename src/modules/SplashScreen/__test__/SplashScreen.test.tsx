import * as React from 'react';
import { shallow } from 'enzyme';
import SplashScreen from '../SplashScreen';

describe('Splash Screen', () => {
    it('splashscreen should match with the snapshot', () => {
        const props = {
            isLoggedIn: false,
            componentId: 'splashscreen',
        };

        const wrapper = shallow(<SplashScreen {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
