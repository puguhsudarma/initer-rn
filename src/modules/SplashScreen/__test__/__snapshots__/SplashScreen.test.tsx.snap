// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen splashscreen should match with the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#17c0d3",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;
